### PR TITLE
fix: fixes version check for Artifact Hub generation

### LINKED ISSUE #331: fix version check in Artifact Hub generation
Ref - https://github.com/open-policy-agent/gatekeeper-library/pull/305#issuecomment-1507529116

### PR DESCRIPTION
**Pull request by nilekhc**
**What this PR does / why we need it**:

Currently, we are checking the template hash from a local file. This requires a constant version upgrade when we add new template files and incrementally update it before merging. This is incorrect since for a new file, the version would start at `1.0.0`.

With this fix, we are now validating the hash of the template file from GitHub to the hash of the local template file. So when a contributor is trying to create a new template file, it will skip the version check and perform the same for an existing template file for a potential version bump.

**Which issue(s) does this PR fix** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #331 

**Special notes for your reviewer**:

### CODE REVIEW

File: scripts/artifacthub/hub.go [hunk 1]
```diff
@@ -221,11 +224,38 @@ func addArtifactHubMetadata(sourceDirectory, destinationPath, ahBasePath string,
...
+	// compare hash with template.yaml in github
+	githubTemplateURL := sourceURL + githubSourceRelativePath
+	resp, err := http.Get(githubTemplateURL)
+	if err != nil {
+		fmt.Println("error while getting constraint template from github")
+		panic(err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		fmt.Printf("constraint template %s not found in github. It is likely that constraint template is being updated locally and not merged to github yet.\n", githubSourceRelativePath)
+
```
Comment (apeabody) [inline scripts/artifacthub/hub.go:237]:
```suggestion
```

File: scripts/artifacthub/hub.go [hunk 2]
```diff
@@ -221,11 +224,38 @@ func addArtifactHubMetadata(sourceDirectory, destinationPath, ahBasePath string,
...
+	githubTemplateURL := sourceURL + githubSourceRelativePath
+	resp, err := http.Get(githubTemplateURL)
+	if err != nil {
+		fmt.Println("error while getting constraint template from github")
+		panic(err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		fmt.Printf("constraint template %s not found in github. It is likely that constraint template is being updated locally and not merged to github yet.\n", githubSourceRelativePath)
+		
+		return nil
```
Comment (apeabody) [inline scripts/artifacthub/hub.go:238]: This test probably needs to be updated? https://github.com/open-policy-agent/gatekeeper-library/blob/master/scripts/artifacthub/hub_test.go#LL184C24-L184C24

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks @nilekhc!

I think `hub_test.go` maybe missing when an expected error does not occur (`err == nil && tc.expectedError`)? https://github.com/open-policy-agent/gatekeeper-library/blob/master/scripts/artifacthub/hub_test.go#L215

File: scripts/artifacthub/hub.go (hunk 1 above)
Comment (nilekhc) [reply scripts/artifacthub/hub.go]: I usually put the `return` on the new line for better code readability. LMK if you have any opinions about the same.

File: scripts/artifacthub/hub.go (hunk 2 above)
Comment (nilekhc) [reply scripts/artifacthub/hub.go:250]: I updated the entire test method.

**Comment by nilekhc** [comment]
> Thanks @nilekhc!
> 
> I think `hub_test.go` maybe missing when an expected error does not occur (`err == nil && tc.expectedError`)? https://github.com/open-policy-agent/gatekeeper-library/blob/master/scripts/artifacthub/hub_test.go#L215

@apeabody I updated the entire test method. PTAL.

**Review by apeabody — APPROVED** [review]
Thanks @nilekhc LGTM!

**Pull request merged**